lib/storage/src/types.rs

```diff
@@ -264,3 +266,25 @@ impl Anonymize for ClusterStatus {
         }
     }
 }
+
+/// Metadata describing extra properties for each peer
+#[derive(Debug, Hash, Serialize, Deserialize, Clone, Eq, PartialEq)]
+pub struct PeerMetadata {
+    /// Peer Qdrant version string
+    version: Option<String>,
+}
```

Comment:
Do we also want to keep metadata here for unknown fields, potentially from newer versions?